### ISSUE TITLE
qalc segfaults upon exit

### ISSUE THREAD

**Issue by BSDKaffee** [Qalculate/libqalculate#38]
qalc is segfaulting upon exit if no commands are issued other than 'help'. If qalc is started and a calculation is performed, it seems to exit normally. Backtrace is from version 1.0.0.

Platform:
FreeBSD 11.0 i386 (clang 3.8.0) 

Steps to reproduce:
1. start qalc
2. enter help command (optional)
3. enter exit command
```
Reading symbols from qalc...done.
(gdb) run
Starting program: /usr/local/bin/qalc 
[New LWP 100449 of process 9739]
> exit

Thread 2 received signal SIGSEGV, Segmentation fault.
[Switching to LWP 100449 of process 9739]
MathStructure::type (this=0x0) at MathStructure.cc:9923
9923            return m_type;
(gdb) bt
#0  MathStructure::type (this=0x0) at MathStructure.cc:9923
#1  0x286fde9c in MathStructure::set (this=0xbbbfde88, o=..., merge_precision=false) at MathStructure.cc:273
#2  0x286fdd5b in MathStructure::MathStructure (this=0xbbbfde88, o=...) at MathStructure.cc:206
#3  0x08112a19 in ViewThread::run (this=0x294e9ce0) at qalc.cc:2828
#4  0x28854c3a in Thread::doRun (data=0x294e9ce0) at util.cc:949
#5  0x282caf2c in ?? () from /lib/libthr.so.3
#6  0x00000000 in ?? ()
```

**Comment by hanna-kn** [comment]
I am unfortunately not able to reproduce the error on Linux with clang 4.0.1. I have a few ideas but it will be difficult for me to test them myself.

Here is a patch that you can try: 
https://gist.githubusercontent.com/hanna-kn/bf5fb12f95d38188737fa377bb76257d/raw/c63c882920faa7c8aa0311fd8b906bc54cf3ce42/38.patch

**Comment by BSDKaffee** [comment]
That seems to fix the issue. Thank you!

**Comment by BSDKaffee** [comment]
Issue is fixed in 2.0.0, so I'm closing this. Thanks again!
